examples/common_vmseries/main.tf

```diff
@@ -314,43 +315,59 @@ module "vmseries" {
   ]
 }
 
+# Create Application Gateway
+
+locals {
+  nics_with_appgw_key = flatten([
+    for k, v in var.vmseries : [
+      for nic in v.interfaces : {
+        vm_key    = k
+        nic_name  = nic.name
+        appgw_key = nic.application_gateway_key
+      } if nic.application_gateway_key != null
+  ]])
+
+  ips_4_nics_with_appgw_key = {
+    for v in local.nics_with_appgw_key :
+    v.appgw_key => module.vmseries[v.vm_key].interfaces["${var.name_prefix}${v.nic_name}"].private_ip_address...
```

Comment:
If it will be added to the provider in the future we should definitely introduce it. Regarding putting a link to the issue in the comments, I'm not really sure. We do it anywhere else?

Reply:
ok, let's leave it as it is.